Options/Options.lua

```diff
@@ -307,30 +307,91 @@ local acOptions = {
 				end
 			end,
 		},
-		separator = {
+		headerBossSettings = {
 			type = "header",
-			name = "",
-			order = 18.2,
+			name = L.headerBossSettings,
+			order = 19,
 		},
 		hideBossWarnings = {
 			type = "toggle",
 			name = L.hideBossWarnings,
-			order = 19,
+			order = 19.1,
 			set = function(info, value)
 				lit.db.hideBossWarnings = value
 			end,
 		},
+		headerBarVilibility = {
+			type = "header",
+			name = L.headerBarVilibility,
+			order = 20,
+		},
 		hideInRaid = {
 			type = "toggle",
 			name = L.hideInRaid,
-			order = 20,
+			order = 20.1,
 			set = function(info, value)
 				lit.db.hideInRaid = value
+				if (value) and (select(2, GetInstanceInfo()) == "raid") then
+					lit:Hide();
+				elseif (not value) and (select(2, GetInstanceInfo()) == "raid") then
+					lit:Show();
+				end
+			end,
+		},
+		hideInInstance = {
+			type = "toggle",
+			name = L.hideInInstance,
+			order = 20.2,
+			set = function(info, value)
+				lit.db.hideInInstance = value
+				if (value) and (select(2, GetInstanceInfo()) == "party") then
+					lit:Hide();
+				elseif (not value) and (select(2, GetInstanceInfo()) == "party") then
+					lit:Show();
+				end
+			end,
+		},
+		hideInArena = {
+			type = "toggle",
+			name = L.hideInArena,
+			order = 20.3,
+			set = function(info, value)
+				lit.db.hideInArena = value
+				if (value) and (select(2, GetInstanceInfo()) == "arena") then
+					lit:Hide();
+				elseif (not value) and (select(2, GetInstanceInfo()) == "arena") then
+					lit:Show();
+				end
+			end,
+		},
+		hideInBattleground = {
+			type = "toggle",
+			name = L.hideInBattleground,
+			order = 20.4,
+			set = function(info, value)
+				lit.db.hideInBattleground = value
+				if (value) and (select(2, GetInstanceInfo()) == "pvp") then
+					lit:Hide();
+				elseif (not value) and (select(2, GetInstanceInfo()) == "pvp") then
+					lit:Show();
+				end
+			end,
+		},
+		hideInScenario  = {
```

Comment:
There is no need for an option to hide in scenario, there are no relevant scenarios currently.